geomstats/geometry/special_linear.py

```diff
@@ -0,0 +1,211 @@
+"""Class for the group of special linear matrices."""
+
+import geomstats.backend as gs
+from geomstats.geometry.invariant_metric import InvariantMetric
+from geomstats.geometry.lie_group import MatrixLieGroup
+from geomstats.geometry.lie_algebra import MatrixLieAlgebra
+from geomstats.geometry.general_linear import GeneralLinear
+
+
+class SpecialLinear(MatrixLieGroup):
+    """Class for the Special Linear group SL(n).
+
+    This is the space of invertible matrices of size n and unit determinant.
+
+    Parameters
+    ----------
+    n : int
+        Integer representing the shape of the matrices: n x n.
+    """
+
+    def __init__(self, n):
+        super(SpecialLinear, self).__init__(
+            dim=int((n * (n - 1)) / 2),
+            n=n,
+            lie_algebra=SpecialLinearLieAlgebra(n=n),
+        )
+
+        self.metric = InvariantMetric(self)
+
+    def belongs(self, point, atol=gs.atol):
+        """Evaluate if a point belongs to the group.
+
+        Check the size and the value of the determinant.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n, n]
+            Point to evaluate.
+        atol : float
+            Absolute tolerance.
+            Optional, default: backend atol.
+
+        Returns
+        -------
+        belongs : array-like, shape=[...,]
+            Boolean evaluating if point belongs to the manifold.
+        """
+        has_right_shape = self.lie_algebra.ambient_space.belongs(point)
+
+        if gs.all(has_right_shape):
+            # ???: det can be negative?
+            return gs.isclose(gs.abs(gs.linalg.det(point)), 1.0, atol=atol)
```

Comment:
No this space is restricted to positive determinant

Reply:
return gs.isclose(gs.linalg.det(point), 1.0, atol=atol)